test/test_other.py

```diff
@@ -8545,7 +8544,7 @@ def test_binaryen_warn_mem(self):
     self.run_process([EMCC, test_file('hello_world.c'), '-sINITIAL_MEMORY=' + str(16 * 1024 * 1024), '--pre-js', 'pre.js', '-sWASM_ASYNC_COMPILATION=0', '-sIMPORTED_MEMORY'])
     out = self.run_js('a.out.js', assert_returncode=NON_ZERO)
     self.assertContained('LinkError', out)
-    self.assertContained("memory import 2 has a larger maximum size 800 than the module's declared maximum", out)
+    self.assertContained("memory import 1 has a larger maximum size 800 than the module's declared maximum", out)
```

Comment:
@kripken in this test the report of which memory gets the error seems pretty nondeterministic. I keep having to adjust it with these feature PRs (it was 2, when I ran it locally before it was 1, and on CI here it looks like it's 0). Do you have any idea what might be causing that?

Reply:
Hmm, no idea, that is very odd. The Binaryen logic for writing out the imports is deterministic:

https://github.com/WebAssembly/binaryen/blob/0b54d74c7ae7e81035a41a4710dca82df19b8638/src/wasm/wasm-binary.cpp#L326

But tags, globals, and functions go before memories, so if memory imports change - like when we move a function from JS to wasm - then I can see that index changing.

But as this isn't what we want to test here, perhaps we can just shorten the check?
```suggestion
    self.assertContained("has a larger maximum size 800 than the module's declared maximum", out)
```

Reply:
What does `memory import 1` mean?   Would it not make sense to use the memory index here?  So it would always be `memory 0 has a ..` (at last in the simple memory case).  Sounds like a binaryen bug, so we can just work around it here of course.

I don't think "import index" is a thing so it wouldn't make sense to use that I think.

Reply:
@sbc100 This error comes from V8, not Binaryen. Trace:
```
a.out.js:684
  var instance = new WebAssembly.Instance(module, info);
                 ^

LinkError: WebAssembly.Instance(): memory import 2 has a larger maximum size 800 than the module's declared maximum 256
    at instantiateSync (a.out.js:684:18)
    at createWasm (a.out.js:737:16)
    at Object.<anonymous> (a.out.js:1122:19)
    at Module._compile (node:internal/modules/cjs/loader:1364:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Module._load (node:internal/modules/cjs/loader:1019:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:128:12)
    at node:internal/main/run_main_module:28:49
Thrown at:
```
But your point is valid, I am not sure why VMs report it that way...

Reply:
Fair enough, probably not worth worrying about, just make the check more lenient.